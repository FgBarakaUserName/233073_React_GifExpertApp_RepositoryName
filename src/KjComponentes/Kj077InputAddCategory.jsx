import React, { useState } from 'react'


export const Kj077InputAddCategory = () => {

  const [kjInputValue, setkjInputValue] = useState("");

  const kjOnInputChange = ( {target} ) => {
    //console.log(target.value)
    setkjInputValue(target.value);
  }

  const kjOnSubmit = (kjEvento2) => {
    kjEvento2.preventDefault();  //para evitar el refresco
    console.log(kjInputValue);
  }

  return (
    <form onSubmit={ (kjEvento2) => kjOnSubmit(kjEvento2) }>
      <input 
        type="text"
        placeholder='kj place holder'
        value={ kjInputValue }
        onChange={ (kjEvento1) => kjOnInputChange(kjEvento1) }
      />
    </form>
  )
}
