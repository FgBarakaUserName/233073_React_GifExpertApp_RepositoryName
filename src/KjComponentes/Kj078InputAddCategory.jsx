import React, { useState } from 'react'

export const Kj078InputAddCategory = ({ kjPropSetArrCat }) => {

  const [kjInputValue, setkjInputValue] = useState("");

  const kjOnInputChange = ( {target} ) => {
    //console.log(target.value)
    setkjInputValue(target.value);
  }

  const kjOnSubmit = (kjEvento2) => {
    kjEvento2.preventDefault();  //para evitar el refresco
    if  (kjInputValue.trim().length <=1) return;
    console.log(kjInputValue);
    kjPropSetArrCat((kjCate)=>[kjInputValue, ...kjCate]);
    setkjInputValue(''); //para inicializar
  }
  
  return (
    <form onSubmit={ (kjEvento2) => kjOnSubmit(kjEvento2) }>
      <input 
        type="text"
        placeholder='kj place holder'
        value={ kjInputValue }
        onChange={ (kjEvento1) => kjOnInputChange(kjEvento1) }
      />
    </form>
  )
}

