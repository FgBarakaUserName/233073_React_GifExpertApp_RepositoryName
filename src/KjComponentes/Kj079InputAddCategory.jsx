import React, { useState } from 'react'

export const Kj079InputAddCategory = ( { kjPropFuncOnNewCategory } ) => {

  const [kjInputValue, setkjInputValue] = useState("");

  const kjOnEveryInputChange = ( {target} ) => {
    //console.log(target.value)
    setkjInputValue(target.value);
  }

  const kjOnOneSubmit = (kjEvento3) => {
    kjEvento3.preventDefault();  //para evitar el refresco
    if  (kjInputValue.trim().length <=1) return;
    //console.log(kjInputValue);
    //aqui ejecuta la funciÃ³n que ha recibido, devolviendo el valor al padre
    kjPropFuncOnNewCategory(kjInputValue.trim())
    setkjInputValue(''); //para inicializar
  }

  /* vas metiendo letras en el INPUT, y va cambiando el KjInputValue
     cuando se da al ENTER, arranca el ONSUBMIT */
  return (
    <form onSubmit={ (kjEvento2) => kjOnOneSubmit(kjEvento2) }> 
      <input 
        type="text"
        placeholder='kj place holder'
        value={ kjInputValue }
        onChange={ (kjEvento1) => kjOnEveryInputChange(kjEvento1) } 
      />
    </form>
  )
}

