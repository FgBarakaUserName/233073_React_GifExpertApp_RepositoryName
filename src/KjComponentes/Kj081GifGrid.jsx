import React from 'react'

export const Kj081GifGrid = ({ kjPropCategoria }) => {

  const kjGifs =[1,2,3];

  return (
    <>
        <h3>{kjPropCategoria}</h3>
        {
           kjGifs.map(( kjDat ) => ( 
                    <p key={kjDat}>
                        {kjDat}
                    </p>
                )
           )
        }
    </>


  )
}
