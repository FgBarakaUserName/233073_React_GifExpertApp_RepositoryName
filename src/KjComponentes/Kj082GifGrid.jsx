import React, { useState, useEffect } from 'react'
import {Kj082GetGifsFromWeb} from '.././KjHelpers/Kj082GetGifsFromWeb.js'

export const Kj082GifGrid = ({ kjPropCategoria }) => {

  const kjGifs =[1,2,3]
  
  useEffect(() => {
    
    Kj082GetGifsFromWeb(kjPropCategoria);
  }, [])
  
  return (
    <>
        <h3>{kjPropCategoria}</h3>
        {
           kjGifs.map(( kjDat ) => ( 
                    <p key={kjDat}>
                        {kjDat}
                    </p>
                )
           )
        }
    </>


  )
}
