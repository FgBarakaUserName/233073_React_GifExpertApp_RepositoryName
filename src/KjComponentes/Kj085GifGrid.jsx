import React, { useState, useEffect } from 'react'
import {Kj082GetGifsFromWeb} from '../KjHelpers/Kj082GetGifsFromWeb.js'

export const Kj085GifGrid = ({ kjPropCategoria }) => {

  const kjGifs =[1,2,3]

  const [kjImagenes, setKjImagenes] = useState([])

  const kjGetImages = async() =>{
     const kjNewImages = await Kj082GetGifsFromWeb(kjPropCategoria);
     setKjImagenes(kjNewImages);
  }  
  
  useEffect(() => {    //no se puede meter el KJ082Get..  porque es asincrono
    kjGetImages()
  }, [])
  
  return (
    <>
        <h3>{kjPropCategoria}</h3>
        {
           kjImagenes.map( ({kjId, kjTitle})  => ( 
            <li key={kjId}>
                {kjTitle}
            </li>
        )
   )
        }
    </>


  )
}
