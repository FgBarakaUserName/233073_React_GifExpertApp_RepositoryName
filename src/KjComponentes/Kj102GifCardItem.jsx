import React from 'react'
import PropTypes from 'prop-types'


// viene de un esparcimiento de propiedades
export const Kj102GifCardItem = ({ kjTitle, kjUrl }) => {
    //console.log(kjTitle);
    return (
        <div className="kjCardItem">
            <img src= { kjUrl} alt= { kjTitle } />
            <p>{ kjTitle }</p>
        </div>
    );
}

Kj102GifCardItem.propTypes ={
    kjTitle  : PropTypes.string.isRequired,
    kjUrl    : PropTypes.string.isRequired
}