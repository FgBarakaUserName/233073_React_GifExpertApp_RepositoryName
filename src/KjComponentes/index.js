

// lecci√≥n 089

export * from './Kj077InputAddCategory';
export * from './Kj078InputAddCategory';
export * from './Kj079InputAddCategory';
export * from './Kj105InputAddCategory';
export * from './Kj106InputAddCategory';

export * from './Kj081GifGrid';
export * from './Kj082GifGrid';
export * from './Kj085GifGrid';
export * from './Kj086GifGrid';
export * from './Kj087GifGrid';
export * from './Kj088GifGrid';
export * from './Kj108GifGrid';

export * from './Kj086GifCardItem';

