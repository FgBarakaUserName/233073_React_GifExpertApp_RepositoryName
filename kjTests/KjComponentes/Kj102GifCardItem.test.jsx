import { render, screen } from "@testing-library/react"
import { Kj102GifCardItem } from "../../src/KjComponentes/Kj102GifCardItem"


describe('de kjxxxGifCardItem ', () => { 

    const kjTitle = "kj mi mama";
    const kjUrl = "https://media1.giphy.com/media/fwePpV0h0KKCjEaXWK/giphy-downsized-medium.gif?cid=c3a7f59cv7ny06c967zom367fknailgeywld0gr0mfv9as44&rid=giphy-downsized-medium.gif&ct=g"
    
    test('kj102 A. debe de hacer match con el snapshot', () => { 
    
        const { container } = render(
                <Kj102GifCardItem kjTitle={kjTitle}  kjUrl= {kjUrl} />
            )
     
        expect( container).toMatchSnapshot();
    })

 })
