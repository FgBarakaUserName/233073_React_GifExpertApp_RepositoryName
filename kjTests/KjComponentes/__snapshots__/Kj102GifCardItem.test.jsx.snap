// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`de kjxxxGifCardItem  kj102 A. debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="kjCardItem"
  >
    <img
      alt="kj mi mama"
      src="https://media1.giphy.com/media/fwePpV0h0KKCjEaXWK/giphy-downsized-medium.gif?cid=c3a7f59cv7ny06c967zom367fknailgeywld0gr0mfv9as44&rid=giphy-downsized-medium.gif&ct=g"
    />
    <p>
      kj mi mama
    </p>
  </div>
</div>
`;
